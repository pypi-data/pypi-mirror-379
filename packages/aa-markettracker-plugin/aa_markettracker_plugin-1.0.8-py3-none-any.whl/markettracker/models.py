from django.db import models
from django.conf import settings
from django.utils.translation import gettext_lazy as _
from django.contrib.auth.models import Group
from allianceauth.authentication.models import CharacterOwnership
from esi.models import Token

# --- opcjonalny fittings ---
try:
    from fittings.models import Fitting
    HAS_FITTINGS = True
except Exception:
    Fitting = None
    HAS_FITTINGS = False


class DiscordWebhook(models.Model):
    """Globalne webhooki dla całego modułu."""
    name = models.CharField(max_length=100, default="Webhook")
    url = models.URLField(unique=True)

    class Meta:
        default_permissions = ()

    def __str__(self) -> str:
        return f"{self.name} – {self.url}"


class DiscordMessage(models.Model):
    """
    Konfiguracja wiadomości i pingów:
    - osobny nagłówek + ping dla Items
    - osobny nagłówek + ping dla Contracts

    Ping można ustawić przez:
      a) item_ping_choice / contract_ping_choice: 'none' | 'here' | 'everyone'
      b) item_ping_group / contract_ping_group: dowolna grupa z AA
    """
    # headers
    item_alert_header = models.CharField(
        max_length=200,
        default="⚠️ MarketTracker Items",
        help_text=_("Header used for items alerts"),
    )
    contract_alert_header = models.CharField(
        max_length=200,
        default="📦 MarketTracker Contracts",
        help_text=_("Header used for contract alerts"),
    )

    # ping target for items
    item_ping_choice = models.CharField(
        max_length=16,
        blank=True,
        null=True,
        help_text=_("Special ping target: none, here, everyone"),
    )
    item_ping_group = models.ForeignKey(
        Group, on_delete=models.SET_NULL, null=True, blank=True,
        help_text=_("Discord role (from AA group) to ping for items"),
        related_name="markettracker_item_pings",
    )

    # ping target for contracts
    contract_ping_choice = models.CharField(
        max_length=16,
        blank=True,
        null=True,
        help_text=_("Special ping target: none, here, everyone"),
    )
    contract_ping_group = models.ForeignKey(
        Group, on_delete=models.SET_NULL, null=True, blank=True,
        help_text=_("Discord role (from AA group) to ping for contracts"),
        related_name="markettracker_contract_pings",
    )

    class Meta:
        default_permissions = ()

    def __str__(self) -> str:
        return "Discord Messages / Pings"


class MarketTrackingConfig(models.Model):
    """Ustawienia śledzenia rynku + progi kolorów."""
    SCOPE_CHOICES = [
        ("region", _("Region")),
        ("structure", _("Structure")),
    ]
    scope = models.CharField(max_length=20, choices=SCOPE_CHOICES, default="region")
    # Tu trzymamy finalne ID (region_id albo structure_id – w zależności od scope)
    location_id = models.BigIntegerField(verbose_name=_("Region or Structure ID"))

    yellow_threshold = models.PositiveIntegerField(
        default=50,
        verbose_name=_("Yellow status threshold (%)")
    )
    red_threshold = models.PositiveIntegerField(
        default=25,
        verbose_name=_("Red status threshold (%)")
    )

    
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        default_permissions = ()

    def __str__(self) -> str:
        return f"{self.scope} – {self.location_id}"


class MarketCharacter(models.Model):
    character = models.OneToOneField(
        CharacterOwnership, on_delete=models.CASCADE, related_name="market_character"
    )
    token = models.OneToOneField(Token, on_delete=models.CASCADE)
    type = models.CharField(max_length=20, default="user")  # "admin" or "user"

    class Meta:
        default_permissions = ()

    def __str__(self):
        return self.character.character.name


class TrackedStructure(models.Model):
    name = models.CharField(max_length=255)
    structure_id = models.BigIntegerField(unique=True)

    class Meta:
        default_permissions = ()

    def __str__(self):
        return self.name


class TrackedItem(models.Model):
    item = models.ForeignKey(
        "eveuniverse.EveType", on_delete=models.CASCADE, verbose_name=_("Item")
    )
    desired_quantity = models.IntegerField(default=0, verbose_name=_("Desired quantity"))
    last_status = models.CharField(max_length=10, default="OK", verbose_name=_("Last status"))
    structure = models.ForeignKey(
        TrackedStructure, on_delete=models.CASCADE, related_name="tracked_items",
        blank=True, null=True
    )

    class Meta:
        unique_together = ("item", "structure")
        default_permissions = ()

    def __str__(self):
        return self.item.name


class MarketOrderSnapshot(models.Model):
    tracked_item = models.ForeignKey(
        "markettracker.TrackedItem",
        on_delete=models.CASCADE,
        related_name="order_snapshots",
    )
    order_id = models.BigIntegerField(unique=True)
    structure_id = models.BigIntegerField()
    price = models.FloatField()
    volume_remain = models.IntegerField()
    is_buy_order = models.BooleanField(default=False)
    issued = models.DateTimeField()

    class Meta:
        default_permissions = ()
        indexes = [
            models.Index(fields=["tracked_item"]),
            models.Index(fields=["structure_id"]),
            models.Index(fields=["is_buy_order"]),
        ]

    def __str__(self):
        return f"{self.tracked_item.item.name} - {self.price}"


class TrackedContract(models.Model):
    class Mode(models.TextChoices):
        CUSTOM = "custom", _("Custom (title match)")
        DOCTRINE = "doctrine", _("Doctrine (fitting match)")

    mode = models.CharField(max_length=16, choices=Mode.choices, default=Mode.CUSTOM)
    title_filter = models.CharField(max_length=120, blank=True)
    fitting = models.ForeignKey(Fitting, null=True, blank=True, on_delete=models.SET_NULL)
    max_price = models.DecimalField(max_digits=20, decimal_places=2, default=0)
    is_active = models.BooleanField(default=True)
    created_by = models.ForeignKey(
        settings.AUTH_USER_MODEL, null=True, blank=True, on_delete=models.SET_NULL, related_name="+"
    )
    created_at = models.DateTimeField(auto_now_add=True)
    desired_quantity = models.PositiveIntegerField(default=0)
    last_status = models.CharField(max_length=10, default="OK")

    class Meta:
        default_permissions = ()
        ordering = ["mode", "title_filter"]

    def display_name(self):
        if self.mode == self.Mode.CUSTOM:
            return self.title_filter or "—"
        return getattr(self.fitting, "name", "—")

    def __str__(self):
        if self.mode == self.Mode.CUSTOM:
            return f"[custom] {self.title_filter or '-'}"
        return f"[doctrine] {getattr(self.fitting, 'name', '—')}"


class ContractSnapshot(models.Model):
    contract_id = models.BigIntegerField(unique=True)
    owner_character_id = models.BigIntegerField()
    owner_character_name = models.CharField(max_length=128, blank=True, default="")
    type = models.CharField(max_length=32, blank=True, default="")
    availability = models.CharField(max_length=32, blank=True, default="")
    status = models.CharField(max_length=32, blank=True, default="")
    title = models.CharField(max_length=255, blank=True, default="")
    date_issued = models.DateTimeField(null=True, blank=True)
    date_expired = models.DateTimeField(null=True, blank=True)
    start_location_id = models.BigIntegerField(null=True, blank=True)
    end_location_id = models.BigIntegerField(null=True, blank=True)
    price = models.FloatField(null=True, blank=True)
    reward = models.FloatField(null=True, blank=True)
    collateral = models.FloatField(null=True, blank=True)
    volume = models.FloatField(null=True, blank=True)
    for_corporation = models.BooleanField(default=False)
    assignee_id = models.BigIntegerField(null=True, blank=True)
    acceptor_id = models.BigIntegerField(null=True, blank=True)
    issuer_id = models.BigIntegerField(null=True, blank=True)
    issuer_corporation_id = models.BigIntegerField(null=True, blank=True)
    items = models.JSONField(default=list, blank=True)
    fetched_at = models.DateTimeField(auto_now=True)

    class Meta:
        default_permissions = ()

    def __str__(self):
        return f"{self.contract_id} ({self.type}, {self.status})"


class ContractError(models.Model):
    class Code(models.TextChoices):
        PRICE_TOO_HIGH = "price", _("Price exceeds max")
        WRONG_HULL = "hull", _("Wrong hull")
        MODULE_MISMATCH = "modules", _("Modules mismatch")
        TITLE_MISMATCH = "title", _("Title mismatch")
        DOCTRINE_MISSING = "doctrine", _("Doctrine fitting missing")

    tracked = models.ForeignKey(TrackedContract, on_delete=models.CASCADE, related_name="errors")
    contract_id = models.BigIntegerField()
    owner_name = models.CharField(max_length=128, blank=True)
    code = models.CharField(max_length=16, choices=Code.choices)
    message = models.TextField(blank=True)
    created_at = models.DateTimeField(auto_now_add=True)
    is_resolved = models.BooleanField(default=False)

    class Meta:
        default_permissions = ()
        ordering = ["-created_at"]


class Delivery(models.Model):
    STATUS_CHOICES = [("PENDING", _("Pending")), ("FINISHED", _("Finished"))]
    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, verbose_name=_("User"))
    character = models.ForeignKey(
        CharacterOwnership, on_delete=models.SET_NULL, null=True, blank=True, verbose_name=_("Character")
    )
    item = models.ForeignKey("eveuniverse.EveType", on_delete=models.CASCADE, verbose_name=_("Item"))
    declared_quantity = models.PositiveIntegerField(verbose_name=_("Declared quantity"))
    delivered_quantity = models.PositiveIntegerField(default=0, verbose_name=_("Delivered quantity"))
    status = models.CharField(max_length=10, choices=STATUS_CHOICES, default="PENDING", verbose_name=_("Status"))
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        default_permissions = ()

    def is_finished(self):
        return self.status == "FINISHED" or self.delivered_quantity >= self.declared_quantity

    def __str__(self):
        return f"{self.user} – {self.item} ({self.delivered_quantity}/{self.declared_quantity})"


class ContractDelivery(models.Model):
    STATUS_CHOICES = [("PENDING", _("Pending")), ("FINISHED", _("Finished"))]
    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, verbose_name=_("User"))
    tracked_contract = models.ForeignKey(
        "markettracker.TrackedContract", on_delete=models.CASCADE, related_name="deliveries", verbose_name=_("Tracked contract")
    )
    declared_quantity = models.PositiveIntegerField(verbose_name=_("Declared quantity"))
    delivered_quantity = models.PositiveIntegerField(default=0, verbose_name=_("Delivered quantity"))
    status = models.CharField(max_length=10, choices=STATUS_CHOICES, default="PENDING", verbose_name=_("Status"))
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        default_permissions = ()

    def is_finished(self):
        return self.status == "FINISHED" or self.delivered_quantity >= self.declared_quantity

    def __str__(self):
        return f"{self.user} – TC#{self.tracked_contract_id} ({self.delivered_quantity}/{self.declared_quantity})"


class General(models.Model):
    """Uprawnienia aplikacji."""
    class Meta:
        managed = False
        default_permissions = ()
        permissions = (
            ("basic_access", "Can access this app"),
            ("can_manage_stocks", "Can set tracked items"),
            ("can_manage_deliveries", "Can see all finished deliveries and delete current ones"),
        )
