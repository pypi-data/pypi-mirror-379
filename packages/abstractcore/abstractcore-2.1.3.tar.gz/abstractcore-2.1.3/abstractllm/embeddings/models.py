"""
Embedding Model Configurations
=============================

SOTA open-source embedding models with optimized configurations.
"""

from dataclasses import dataclass
from typing import Dict, List, Optional
from enum import Enum


class EmbeddingBackend(Enum):
    """Available inference backends for embeddings."""
    PYTORCH = "pytorch"
    ONNX = "onnx"
    OPENVINO = "openvino"


@dataclass
class EmbeddingModelConfig:
    """Configuration for an embedding model."""
    name: str
    model_id: str
    dimension: int
    max_sequence_length: int
    supports_matryoshka: bool
    matryoshka_dims: Optional[List[int]]
    description: str
    multilingual: bool = False
    size_mb: Optional[float] = None


# SOTA 2025 embedding models
EMBEDDING_MODELS: Dict[str, EmbeddingModelConfig] = {
    "embeddinggemma": EmbeddingModelConfig(
        name="embeddinggemma",
        model_id="google/embeddinggemma-300m",
        dimension=768,
        max_sequence_length=8192,
        supports_matryoshka=True,
        matryoshka_dims=[768, 512, 256, 128],
        description="Google's 2025 SOTA on-device embedding model (300M params)",
        multilingual=True,
        size_mb=300
    ),
    "granite": EmbeddingModelConfig(
        name="granite",
        model_id="ibm-granite/granite-embedding-278m-multilingual",
        dimension=768,
        max_sequence_length=512,
        supports_matryoshka=False,
        matryoshka_dims=None,
        description="IBM's enterprise-grade multilingual embedding model (278M params)",
        multilingual=True,
        size_mb=278
    ),
    "stella-400m": EmbeddingModelConfig(
        name="stella-400m",
        model_id="dunzhang/stella_en_400M_v5",
        dimension=1024,
        max_sequence_length=512,
        supports_matryoshka=True,
        matryoshka_dims=[1024, 768, 512, 256],
        description="Excellent accuracy-to-size ratio, fine-tunable",
        multilingual=False,
        size_mb=400
    ),
    "nomic-embed": EmbeddingModelConfig(
        name="nomic-embed",
        model_id="nomic-ai/nomic-embed-text-v1.5",
        dimension=768,
        max_sequence_length=8192,
        supports_matryoshka=True,
        matryoshka_dims=[768, 512, 256, 128],
        description="High-quality English embeddings, outperforms text-embedding-ada-002",
        multilingual=False,
        size_mb=550
    ),
    "mxbai-large": EmbeddingModelConfig(
        name="mxbai-large",
        model_id="mixedbread-ai/mxbai-embed-large-v1",
        dimension=1024,
        max_sequence_length=512,
        supports_matryoshka=True,
        matryoshka_dims=[1024, 768, 512, 256],
        description="Outperforms text-embedding-3-large while being smaller",
        multilingual=False,
        size_mb=650
    )
}


def get_model_config(model_name: str) -> EmbeddingModelConfig:
    """Get configuration for a specific model.

    Args:
        model_name: Name of the embedding model

    Returns:
        EmbeddingModelConfig for the specified model

    Raises:
        ValueError: If model_name is not supported
    """
    if model_name not in EMBEDDING_MODELS:
        available = ", ".join(EMBEDDING_MODELS.keys())
        raise ValueError(f"Model '{model_name}' not supported. Available: {available}")

    return EMBEDDING_MODELS[model_name]


def list_available_models() -> List[str]:
    """List all available embedding models."""
    return list(EMBEDDING_MODELS.keys())


def get_default_model() -> str:
    """Get the default embedding model (EmbeddingGemma)."""
    return "embeddinggemma"