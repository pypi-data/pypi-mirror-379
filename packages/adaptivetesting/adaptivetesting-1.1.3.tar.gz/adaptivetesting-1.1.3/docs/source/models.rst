models Module
=============

.. automodule:: adaptivetesting.models
   :members:
   :undoc-members:
   :show-inheritance:
   :imported-members:
   :special-members: __init__