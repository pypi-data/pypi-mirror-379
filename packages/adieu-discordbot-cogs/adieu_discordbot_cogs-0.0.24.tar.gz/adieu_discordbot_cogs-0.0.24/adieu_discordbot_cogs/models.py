"""
App Models
Create your models in here
"""

# Django
from django.db import models


class General(models.Model):
    """Meta model for app permissions"""

    class Meta:
        """Meta definitions"""

        managed = False
        default_permissions = ()
        permissions = (("force_sync", "Can use force sync"), ("admin_commands", "Can use Admin commands"))
