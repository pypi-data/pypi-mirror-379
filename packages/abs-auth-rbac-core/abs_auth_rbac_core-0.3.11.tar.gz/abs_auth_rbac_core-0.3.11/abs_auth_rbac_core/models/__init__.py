from .permissions import Permission
from .roles import Role
from .user_role import UserRole
from .user import Users
from .role_permission import RolePermission
from .user_permission import UserPermission

__all__ = ["Permission", "Role", "UserRole", "Users","RolePermission","UserPermission"]