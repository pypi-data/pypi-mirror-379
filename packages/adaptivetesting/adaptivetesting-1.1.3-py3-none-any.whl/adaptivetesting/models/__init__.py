from .__adaptive_test import AdaptiveTest
from .__algorithm_exception import AlgorithmException
from .__item_pool import ItemPool 
from .__item_selection_exception import ItemSelectionException
from .__test_item import TestItem
from .__test_result import TestResult
from .__misc import ResultOutputFormat, StoppingCriterion
