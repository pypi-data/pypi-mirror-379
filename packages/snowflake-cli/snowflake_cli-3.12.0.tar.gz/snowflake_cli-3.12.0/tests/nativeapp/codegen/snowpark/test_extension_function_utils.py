# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from textwrap import dedent

import pytest
import snowflake.cli._plugins.nativeapp.codegen.snowpark.extension_function_utils as ef_utils
from snowflake.cli._plugins.nativeapp.codegen.snowpark.models import (
    NativeAppExtensionFunction,
)
from snowflake.cli.api.project.schemas.v1.snowpark.argument import Argument


@pytest.mark.parametrize(
    "function_type, expected",
    [
        ("procedure", "PROCEDURE"),
        ("function", "FUNCTION"),
        ("aggregate function", "AGGREGATE FUNCTION"),
        ("table function", "FUNCTION"),
    ],
)
def test_get_sql_object_type(
    function_type, expected, native_app_extension_function_raw_data
):
    native_app_extension_function_raw_data["type"] = function_type
    extension_fn = NativeAppExtensionFunction(**native_app_extension_function_raw_data)
    assert ef_utils.get_sql_object_type(extension_fn) == expected


def test_get_sql_argument_signature():
    arg = Argument(name="foo", type="int")
    assert ef_utils.get_sql_argument_signature(arg) == "foo int"

    arg = Argument(name="foo", type="int", default="42")
    assert ef_utils.get_sql_argument_signature(arg) == "foo int DEFAULT 42"


def test_get_qualified_object_name(native_app_extension_function):
    native_app_extension_function.name = "foo"
    native_app_extension_function.schema_name = None

    assert ef_utils.get_qualified_object_name(native_app_extension_function) == "foo"

    native_app_extension_function.name = "foo"
    native_app_extension_function.schema_name = "my_schema"

    assert (
        ef_utils.get_qualified_object_name(native_app_extension_function)
        == "my_schema.foo"
    )

    native_app_extension_function.name = "foo"
    native_app_extension_function.schema_name = "my schema"

    assert (
        ef_utils.get_qualified_object_name(native_app_extension_function)
        == '"my schema".foo'
    )

    native_app_extension_function.name = "foo"
    native_app_extension_function.schema_name = "my.full.schema"

    assert (
        ef_utils.get_qualified_object_name(native_app_extension_function)
        == "my.full.schema.foo"
    )

    native_app_extension_function.name = "foo"
    native_app_extension_function.schema_name = "my.full schema.with special chars"

    assert (
        ef_utils.get_qualified_object_name(native_app_extension_function)
        == 'my."full schema"."with special chars".foo'
    )


def test_get_function_type_signature_for_grant(native_app_extension_function):
    assert (
        ef_utils.get_function_type_signature_for_grant(native_app_extension_function)
        == "int"
    )

    native_app_extension_function.signature = []
    assert (
        ef_utils.get_function_type_signature_for_grant(native_app_extension_function)
        == ""
    )

    native_app_extension_function.signature = [
        Argument(name="foo", type="int", default="42"),
        Argument(name="bar", type="varchar"),
    ]
    assert (
        ef_utils.get_function_type_signature_for_grant(native_app_extension_function)
        == "int, varchar"
    )


def test_ensure_string_literal():
    assert ef_utils.ensure_string_literal("") == "''"
    assert ef_utils.ensure_string_literal("abc") == "'abc'"
    assert ef_utils.ensure_string_literal("'abc'") == "'abc'"
    assert ef_utils.ensure_string_literal("'abc def'") == "'abc def'"
    assert ef_utils.ensure_string_literal("'abc") == r"'\'abc'"
    assert ef_utils.ensure_string_literal("abc'") == r"'abc\''"


def test_ensure_all_string_literals():
    assert ef_utils.ensure_all_string_literals([]) == []
    assert ef_utils.ensure_all_string_literals(["", "foo", "'bar'"]) == [
        "''",
        "'foo'",
        "'bar'",
    ]


TEST_SNOWPARK_CODE = dedent(
    """
import snowflake.snowpark

from typing import Optional
from snowflake.snowpark.session import Session
from snowflake.snowpark.functions import col, sum, sproc


@custom
@sproc(native_app_params={'schema': 'math', 'application_roles': ['app_public', 'app_admin']})
def sproc_sum(session: Session, first: int, second: int) -> int:
    return first + second

@udf(native_app_params={
    'schema': 'math',
    'application_roles': ['app_public', 'app_admin']
})
@module.annotation
def udf_sum(first: int, second: int) -> int:
    return first + second
    
@custom
def helper():
    pass
    
    
@custom
@udtf(
    name="alt_int",
    replace=True,
    output_schema=StructType([StructField("number", IntegerType())]),
    input_types=[IntegerType()],
)
@module.annotation
class Alternator:
    def __init__(self):
        self._positive = True

    def process(self, n):
        for i in range(n):
            if self._positive:
                yield (1,)
            else:
                yield (-1,)
            self._positive = not self._positive
"""
).strip()


def test_deannotate_module_source_removes_all_annotations(os_agnostic_snapshot):
    sproc = NativeAppExtensionFunction(
        type="procedure",
        handler="math_fns.sproc_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.udf_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udtf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.Alternator",
        returns="TABLE",
        signature=[
            Argument(name="n", type="int"),
        ],
    )
    assert (
        ef_utils.deannotate_module_source(TEST_SNOWPARK_CODE, [sproc, udf, udtf])
        == os_agnostic_snapshot
    )


def test_deannotate_module_source_preserves_specified_annotations(os_agnostic_snapshot):
    sproc = NativeAppExtensionFunction(
        type="procedure",
        handler="math_fns.sproc_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.udf_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udtf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.Alternator",
        returns="TABLE",
        signature=[
            Argument(name="n", type="int"),
        ],
    )
    assert (
        ef_utils.deannotate_module_source(
            TEST_SNOWPARK_CODE, [sproc, udf, udtf], annotations_to_preserve=["custom"]
        )
        == os_agnostic_snapshot
    )
    assert (
        ef_utils.deannotate_module_source(
            TEST_SNOWPARK_CODE,
            [sproc, udf, udtf],
            annotations_to_preserve=["module.annotation"],
        )
        == os_agnostic_snapshot
    )


def test_deannotate_module_source_is_identity_when_no_functions_present(
    os_agnostic_snapshot,
):
    sproc = NativeAppExtensionFunction(
        type="procedure",
        handler="math_fns.sproc_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.udf_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udtf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.Alternator",
        returns="TABLE",
        signature=[
            Argument(name="n", type="int"),
        ],
    )

    non_annotated_code = dedent(
        """
    if __name__ == "__main__":
        print("All your base are belong to us")
    """
    )

    assert (
        ef_utils.deannotate_module_source(
            non_annotated_code, [sproc, udf, udtf], annotations_to_preserve=["custom"]
        )
        == non_annotated_code
    )


def test_deannotate_module_source_is_identity_when_no_annotated_functions_present(
    os_agnostic_snapshot,
):
    sproc = NativeAppExtensionFunction(
        type="procedure",
        handler="math_fns.sproc_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.udf_sum",
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udtf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.Alternator",
        returns="TABLE",
        signature=[
            Argument(name="n", type="int"),
        ],
    )

    non_annotated_code = dedent(
        """
    def foo():
        pass
        
    class Bar:
        pass
    """
    )

    assert (
        ef_utils.deannotate_module_source(
            non_annotated_code, [sproc, udf, udtf], annotations_to_preserve=["custom"]
        )
        == non_annotated_code
    )


def test_deannotate_module_source_is_identity_when_extension_function_does_not_match(
    os_agnostic_snapshot,
):
    sproc = NativeAppExtensionFunction(
        type="procedure",
        handler="math_fns.my_sproc_sum",  # trigger mismatch
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.my_udf_sum",  # trigger mismatch
        returns="int",
        signature=[
            Argument(name="first", type="int"),
            Argument(name="second", type="int"),
        ],
    )
    udtf = NativeAppExtensionFunction(
        type="function",
        handler="math_fns.MyAlternator",  # trigger mismatch
        returns="TABLE",
        signature=[
            Argument(name="n", type="int"),
        ],
    )

    assert (
        ef_utils.deannotate_module_source(TEST_SNOWPARK_CODE, [sproc, udf, udtf])
        == TEST_SNOWPARK_CODE
    )
