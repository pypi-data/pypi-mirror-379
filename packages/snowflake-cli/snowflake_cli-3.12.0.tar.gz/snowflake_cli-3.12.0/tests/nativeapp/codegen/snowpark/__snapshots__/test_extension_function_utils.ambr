# serializer version: 1
# name: test_deannotate_module_source_preserves_specified_annotations
  '''
  import snowflake.snowpark
  
  from typing import Optional
  from snowflake.snowpark.session import Session
  from snowflake.snowpark.functions import col, sum, sproc
  
  
  @custom
  #: @sproc(native_app_params={'schema': 'math', 'application_roles': ['app_public', 'app_admin']})
  def sproc_sum(session: Session, first: int, second: int) -> int:
      return first + second
  
  #: @udf(native_app_params={
  #:     'schema': 'math',
  #:     'application_roles': ['app_public', 'app_admin']
  #: })
  #: @module.annotation
  def udf_sum(first: int, second: int) -> int:
      return first + second
  
  @custom
  def helper():
      pass
  
  
  @custom
  #: @udtf(
  #:     name="alt_int",
  #:     replace=True,
  #:     output_schema=StructType([StructField("number", IntegerType())]),
  #:     input_types=[IntegerType()],
  #: )
  #: @module.annotation
  class Alternator:
      def __init__(self):
          self._positive = True
  
      def process(self, n):
          for i in range(n):
              if self._positive:
                  yield (1,)
              else:
                  yield (-1,)
              self._positive = not self._positive
  '''
# ---
# name: test_deannotate_module_source_preserves_specified_annotations.1
  '''
  import snowflake.snowpark
  
  from typing import Optional
  from snowflake.snowpark.session import Session
  from snowflake.snowpark.functions import col, sum, sproc
  
  
  #: @custom
  #: @sproc(native_app_params={'schema': 'math', 'application_roles': ['app_public', 'app_admin']})
  def sproc_sum(session: Session, first: int, second: int) -> int:
      return first + second
  
  #: @udf(native_app_params={
  #:     'schema': 'math',
  #:     'application_roles': ['app_public', 'app_admin']
  #: })
  @module.annotation
  def udf_sum(first: int, second: int) -> int:
      return first + second
  
  @custom
  def helper():
      pass
  
  
  #: @custom
  #: @udtf(
  #:     name="alt_int",
  #:     replace=True,
  #:     output_schema=StructType([StructField("number", IntegerType())]),
  #:     input_types=[IntegerType()],
  #: )
  @module.annotation
  class Alternator:
      def __init__(self):
          self._positive = True
  
      def process(self, n):
          for i in range(n):
              if self._positive:
                  yield (1,)
              else:
                  yield (-1,)
              self._positive = not self._positive
  '''
# ---
# name: test_deannotate_module_source_removes_all_annotations
  '''
  import snowflake.snowpark
  
  from typing import Optional
  from snowflake.snowpark.session import Session
  from snowflake.snowpark.functions import col, sum, sproc
  
  
  #: @custom
  #: @sproc(native_app_params={'schema': 'math', 'application_roles': ['app_public', 'app_admin']})
  def sproc_sum(session: Session, first: int, second: int) -> int:
      return first + second
  
  #: @udf(native_app_params={
  #:     'schema': 'math',
  #:     'application_roles': ['app_public', 'app_admin']
  #: })
  #: @module.annotation
  def udf_sum(first: int, second: int) -> int:
      return first + second
  
  @custom
  def helper():
      pass
  
  
  #: @custom
  #: @udtf(
  #:     name="alt_int",
  #:     replace=True,
  #:     output_schema=StructType([StructField("number", IntegerType())]),
  #:     input_types=[IntegerType()],
  #: )
  #: @module.annotation
  class Alternator:
      def __init__(self):
          self._positive = True
  
      def process(self, n):
          for i in range(n):
              if self._positive:
                  yield (1,)
              else:
                  yield (-1,)
              self._positive = not self._positive
  '''
# ---
